gef.py

```diff
@@ -8291,18 +8294,46 @@ def do_invoke(self, argv):
         for entry in vmmap:
             if argv and not argv[0] in entry.path:
                 continue
-            l = []
-            l.append(format_address(entry.page_start))
-            l.append(format_address(entry.page_end))
-            l.append(format_address(entry.offset))
 
-            if entry.permission.value == (Permission.READ|Permission.WRITE|Permission.EXECUTE) :
-                l.append(Color.colorify(str(entry.permission), "bold red"))
-            else:
-                l.append(str(entry.permission))
+            self.print_entry(entry)
+        return
 
-            l.append(entry.path)
-            gef_print(" ".join(l))
+    def print_entry(self, entry):
+        line_attributes = []
+        if entry.path == "[stack]":
+            line_attributes.append(get_gef_setting("theme.address_stack"))
+        elif entry.path == "[heap]":
+            line_attributes.append(get_gef_setting("theme.address_heap"))
+        elif entry.permission.value & Permission.READ and entry.permission.value & Permission.EXECUTE:
+            line_attributes.append(get_gef_setting("theme.address_code"))
+
+        line_attributes = " ".join(line_attributes)
```

Comment:
Why append to an array then join, instead of just assigning to a string? Like the following:

```py
        line_attributes = ""
        if entry.path == "[stack]":
            line_attributes = get_gef_setting("theme.address_stack")
        elif entry.path == "[heap]":
            line_attributes = get_gef_setting("theme.address_heap")
        elif entry.permission.value & Permission.READ and entry.permission.value & Permission.EXECUTE:
            line_attributes = get_gef_setting("theme.address_code")
```

Reply:
No reason anymore. I'll change that.

Reply:
we used to do it this way because it made things noticeably faster (fewer allocations)

Reply:
Are you referring to the lines below it (l.append...)? Wouldn't there be only 1 allocation (the new color) for the section highlighted above?